packages/block-editor/src/components/block-list/block.js

```diff
@@ -522,7 +519,7 @@ function BlockListBlock( {
 			onClick={ onTouchStop }
 			onKeyDown={ onKeyDown }
 			tabIndex="0"
-			aria-label={ blockLabel }
+			aria-label={ blockAriaLabel }
```

Comment:
This might represent a bit of a performance issue for something like the paragraph block where the aria-label contains all the paragraph text. Typing in the block also updates the aria-label!

An option might be to debounce the function.

Reply:
> 
> 
> This might represent a bit of a performance issue for something like the paragraph block where the aria-label contains all the paragraph text. Typing in the block also updates the aria-label!

How do the mobile friends do it? An option might be to update the paragraph text, for example, once editing mode is exited.

Reply:
I have a feeling it's not as much of an issue on the mobile side as there's no document object model. I've updated the code now so that the function is debounced, hopefully that's an ok solution.